components/script/dom/xrsession.rs

```diff
@@ -510,7 +518,13 @@ impl XRSession {
 
     /// <https://immersive-web.github.io/webxr/#xrframe-apply-frame-updates>
     fn apply_frame_updates(&self, _frame: &XRFrame) {
-        // TODO: add a comment about why this is empty right now!
+        // <https://www.w3.org/TR/webxr-gamepads-module-1/#xrframe-apply-gamepad-frame-updates>
+        for (id, frame) in &*self.input_frames.borrow() {
```

Comment:
Should we use borrow_mut() and drain here instead? It would avoid the clone call a few lines down, and would avoid any problems where no updated input was received in a subsequent frame but we still process the previous update anyways.